[gstreamer][gst-rtsp-server] Support Linux, fix OSX build and fix pkgconfig

JackBoosY opened the pull request:
Add Linux support, fix OSX build and fix gstreamer components's pkgconfig files.

Fixes #20785 #20812.
Related PR: https://github.com/microsoft/vcpkg/pull/20555.

Commit:
[gstreamer] Support Linux and fix pkgconfig

Commit:
version

Commit:
remove baseline record

Commit:
Fix installation

Commit:
version

Commit:
[gst-rtsp-server] Fix linux build

Commit:
version

Commit:
Add qtmultimedia:x64-linux to fail

Commit:
Test osx build

Commit:
Revert test code

Commit:
Revert test code

Commit:
version

Commit:
Revert test code

Commit:
version

joiskash commented:
Are there any updates on this? we have a dependency on it and can help out if required.

joiskash commented:
Also the information in browse packages  on vcpkg.io is misleading it says x64-linux is supported
![image](https://user-images.githubusercontent.com/48797364/151771921-a8357d50-7e3f-497c-a8fa-8a45daf24dc4.png)

BillyONeal commented:
> Are there any updates on this? we have a dependency on it and can help out if required.

Given that upstream has indicated they do not like the way the current port is packaged and fixing it to be consistent with what they want requires a major overhaul, I don't think there are updates here.

joiskash commented:
> Also the information in browse packages on vcpkg.io is misleading it says x64-linux is supported ![image](https://user-images.githubusercontent.com/48797364/151771921-a8357d50-7e3f-497c-a8fa-8a45daf24dc4.png)

Also any answer to this? It seems odd that the vcpkg.json support field of being unable to support certain platforms does not show up on the vcpkg.io

BillyONeal commented:
> Also any answer to this? It seems odd that the vcpkg.json support field details of the platforms does not show up on the vcpkg.io

vcpkg.io is a best guess approximation based on what `supports` says, not based on "what actually succeeded in use".

Commit:
Merge branch 'master' of https://github.com/microsoft/vcpkg into dev/jack/20785

Commit:
version

JackBoosY commented:
@luncliff I'm going to close this PR, please feel free to use my branch to create a new one.

Thanks.

luncliff committed:
[gstreamer] build with 1.19.3 (#4)

* [gstreamer] exclude plugin sources from build

* 1.19.2 version doesn't contain plugins folder for good/bad/ugly.
  Bump to 1.19.3 which has subprojects
* Remove 'gstreamer-1.0' from install path.

There might be a option for plugin scan...
Need to check docs for it

* [gstreamer] update vcpkg.json

* remove gstreamer failure in ci.baseline.txt

* [gstreamer] remove release options

* [gstreamer] fix PDB, tool install in Windows

* [gstreamer] remove 'gstreamer-1.0' from install path

* [gstreamer] remove unused patches

* [gstreamer] fix mp3lame link error in windows

* [gstreamer] resurrect gstglconfig.h

* [gstreamer] revert pkgconfig install path

* [gstreamer] rewrite port/patch

* [gstreamer] change pkgconfig fixup

* [gstreamer] fix syntax error after merge

* Update ports/gstreamer/portfile.cmake

Co-authored-by: Jack·Boos·Yu <47264268+JackBoosY@users.noreply.github.com>

* Update ports/gstreamer/portfile.cmake

Co-authored-by: Jack·Boos·Yu <47264268+JackBoosY@users.noreply.github.com>

* [gstremaer] fix license install

* update version JSON

Co-authored-by: Jack·Boos·Yu <47264268+JackBoosY@users.noreply.github.com>

JackBoosY commented:
@luncliff I noticed that gstream's `supports` value was changed to `!linux & !uwp` in `vcpkg.json` now, should I sync it?

luncliff commented:
@JackBoosY Yeah. I just read #23125. `port-version` 4 looks better than current one

luncliff commented:
I think we can just use [`vcpkg.json` of the `master` branch](https://github.com/JackBoosY/vcpkg/blob/master/ports/gstreamer/vcpkg.json) with `version` change.

Commit:
Merge branch 'master' of https://github.com/microsoft/vcpkg into dev/jack/20785

Commit:
Merge branch 'dev/jack/20785' of https://github.com/JackBoosY/vcpkg into dev/jack/20785

JackBoosY commented:
@luncliff Since I merged my branch with master, can you please review again?

Thanks.

luncliff commented:
> @luncliff Since I merged my branch with master, can you please review again?
> 
> Thanks.

With https://github.com/JackBoosY/vcpkg/pull/5, it looks good to me.

luncliff committed:
[gstreamer] fix feature option 'flac' (#5)

* x264 requires GPL license

PhoebeHui commented on ports/gstreamer/portfile.cmake:
```diff
@@ -124,61 +42,51 @@ else()
     set(PLUGIN_UGLY_SUPPORT disabled)
 endif()
 
-if ("gl-graphene" IN_LIST FEATURES)
-    set(GL_GRAPHENE enabled)
-else()
-    set(GL_GRAPHENE disabled)
-endif()
```
Why remove this feature?

luncliff replied:
It seems like my mistake.
https://gitlab.freedesktop.org/gstreamer/gstreamer/-/blob/1.19.3/subprojects/gst-plugins-base/meson_options.txt#L26 provides the value 'auto'.
That was why I couldn't detect the problem...

PhoebeHui commented on ports/gstreamer/portfile.cmake:
```diff
@@ -204,104 +113,92 @@ vcpkg_configure_meson(
         -Dgst-plugins-ugly:orc=disabled
-        -Dgst-plugins-ugly:x264=${PLUGIN_UGLY_X264}
-        # see ${GST_BUILD_SOURCE_PATH}/meson_options.txt
-        -Dpython=disabled
-        -Dlibav=disabled
-        -Dlibnice=disabled # libnice
-        -Ddevtools=disabled
-        -Dges=disabled
-        -Drtsp_server=disabled
-        -Domx=disabled
-        -Dvaapi=disabled
-        -Dsharp=disabled
-        -Drs=disabled
-        -Dgst-examples=disabled
-        -Dtls=disabled
-        -Dtests=disabled    # common options
-        -Dexamples=disabled
-        -Dintrospection=disabled
-        -Dnls=disabled
-        -Dorc=disabled
-        -Ddoc=disabled
-        -Dgtk_doc=disabled
-        -Ddevtools=disabled
```
Does these options deprecated?

luncliff replied:
In short, yes. These options are for `gst-build`, which was a build repository for `gstreamer` until 1.19.2.
Since 1.19.3, the plugins and sub-projects are merged into the mono repository. So providing these options are not required anymore.

luncliff commented on ports/gstreamer/portfile.cmake:
```diff
@@ -187,11 +95,12 @@ vcpkg_configure_meson(
         -Dgst-plugins-base:nls=disabled
         -Dgst-plugins-base:orc=disabled
         -Dgst-plugins-base:pango=disabled
-        -Dgst-plugins-base:gl-graphene=${GL_GRAPHENE}
```
Looks like this option should be resurrected. Sorry.
https://github.com/microsoft/vcpkg/pull/20814/files#r828903103

Commit:
Apply suggestion

Commit:
Merge branch 'master' of https://github.com/microsoft/vcpkg into dev/jack/20785

Commit:
version

JackBoosY commented:
Close and reopen this PR to retrigger the suggestion agent.

JackBoosY committed:
Merge branch 'master' into dev/jack/20785

Commit:
Add supports and dependency features

Commit:
version

Commit:
Merge branch 'master' of https://github.com/microsoft/vcpkg into dev/jack/20785

Commit:
Merge branch 'dev/jack/20785' of https://github.com/JackBoosY/vcpkg into dev/jack/20785

Commit:
Correct the dependencies according to the platform

Commit:
version

Commit:
Add supports, add license

Commit:
version

Commit:
Try to avoid trigger gstreamer[plugins-bad]:x64-osx

Commit:
Update dependencies

Commit:
Restore some changes

Commit:
format manifest

Commit:
version

JackBoosY commented:
The OSX regression fixed by the upstream PR: https://gitlab.freedesktop.org/gstreamer/gstreamer/-/merge_requests/2261
